apps/web/components/v2/eventtype/EventLimitsTab.tsx

```diff
@@ -1,21 +1,31 @@
 import { useAutoAnimate } from "@formkit/auto-animate/react";
 import * as RadioGroup from "@radix-ui/react-radio-group";
 import { EventTypeSetupInfered, FormValues } from "pages/event-types/[type]";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import { useFormContext, Controller, useWatch } from "react-hook-form";
 
 import { classNames } from "@calcom/lib";
+import convertToNewDurationType from "@calcom/lib/convertToNewDurationType";
+import findDurationType from "@calcom/lib/findDurationType";
 import { useLocale } from "@calcom/lib/hooks/useLocale";
 import { PeriodType } from "@calcom/prisma/client";
 import type { BookingLimit } from "@calcom/types/Calendar";
 import { Icon } from "@calcom/ui";
-import { Select, Switch, Label, Input, MinutesField, Button, SettingsToggle } from "@calcom/ui/v2";
+import { Select, Switch, Label, Input, Button, SettingsToggle, InputField } from "@calcom/ui/v2";
 import DateRangePicker from "@calcom/ui/v2/core/form/date-range-picker/DateRangePicker";
 
+export const durationArray: string[] = ["minutes"];
+
 export const EventLimitsTab = (props: Pick<EventTypeSetupInfered, "eventType">) => {
   const { t } = useLocale();
   const formMethods = useFormContext<FormValues>();
   const { eventType } = props;
+  let minimumBookingNoticeType = findDurationType(eventType.minimumBookingNotice);
```

Comment:
I think we should also change this to a ref, since also changing normal variables in a component is not a guarantee that the component wil rerender. Now I'm thinking of it — perhaps we don't even need this to be a changing value anymore as soon as we make the previousValue a ref, since you're only using it to set an initial value and shifting the array (which will be gone when using a ref). Let me know if you want more input from me on this.